src/openms/include/OpenMS/ANALYSIS/ID/TopPerc.h

```diff
@@ -0,0 +1,200 @@
+// ETH Zurich, and Freie Universitaet Berlin 2002-2015.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Mathias Walzer $
+// $Authors: Mathias Walzer, Matthew The $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_ID_TOPPERC_H
+#define OPENMS_ANALYSIS_ID_TOPPERC_H
+
+#include <vector>
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <map>
+#include <algorithm>
+#include <limits>
+
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/KERNEL/StandardTypes.h>
+#include <OpenMS/FORMAT/FileHandler.h>
+#include <OpenMS/DATASTRUCTURES/DataValue.h>
+
+#include <boost/lexical_cast.hpp>
+
+namespace OpenMS
+{
+    /**
+        @brief Percolator feature and I/O handler
+
+        This class contains functions to handle (compute, aggregate) Percolator
+        features.
+    */
+
+    class OPENMS_DLLAPI TopPerc
+    {
+
+    public:
+        /**
+         * @brief concatMULTISEPeptideIds
+         * @param all_peptide_ids PeptideIdentification vector to append to
+         * @param new_peptide_ids PeptideIdentification vector to be appended
+         * @param search_engine search engine to depend on for feature creation
+         *
+         * Appends a vector of PeptideIdentification to another and registers concatenation Percolator features depending on given search engine.
```

Comment:
The whole integration process is complicated and relies on registering the right features for Percolator input facilitation.